### PR TITLE
Add option to disable Matcher errors

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

So that the `AttributeRuler` component runs and doesn't produce cryptic errors when some pipeline components are disabled:

* Add option to disable Matcher errors when a doc doesn't contain a particular type of annotation

Minor additional change:

* Update `AttributeRuler.load_from_morph_rules` to allow direct `MORPH`
values

### Types of change

Enhancement.

## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
I could totally imagine a different setting (a class setting?) or something here. I'm not particularly picky about the details, I just want the functionality because otherwise there are going to be headaches with the default pipelines and the attribute ruler.

**Comment by honnibal** [comment]
I think it makes sense to have a pipeline component that can match on more patterns than are set at one particular time. Like, you should be able to use the same patterns, and see how much more precise the matcher becomes when you have POS tags available vs when you don't. So I support the idea behind this.

`suppress_errors` sounds sort of ominous (and vague) though. How about `allow_missing`?

**Comment by adrianeboyd** [comment]
`missing` is even more vague? I suppose there's no harm in a long option like `allow_missing_annotation`? But that's not even really accurate, since the docs only have to be partially annotated. Hmm...

**Comment by adrianeboyd** [comment]
`skip_annotation_checks`?

**Comment by honnibal** [comment]
I think we should just go with `allow_missing`. I think it's clear and memorable enough, it's not like either documents, patterns or anything else could be "missing". The only thing that can be missing are the values for the attributes. The problem I had with `suppress_errors` is that the user is wondering what type of errors? All errors?

File: spacy/matcher/matcher.pyx [hunk 1]
```diff
@@ -195,7 +195,7 @@ cdef class Matcher:
                 else:
                     yield doc
 
-    def __call__(self, object doclike, *, as_spans=False):
+    def __call__(self, object doclike, *, as_spans=False, suppress_errors=False):
```
Comment (honnibal) [inline spacy/matcher/matcher.pyx:198]:
```suggestion
    def __call__(self, object doclike, *, as_spans=False, allow_missing=False):
```

File: spacy/matcher/matcher.pyx [hunk 2]
```diff
@@ -215,16 +215,17 @@ cdef class Matcher:
...
-            raise ValueError(Errors.E155.format(pipe="tagger", attr="TAG"))
-        if POS in self._seen_attrs and not doc.has_annotation("POS"):
-            raise ValueError(Errors.E155.format(pipe="morphologizer", attr="POS"))
-        if MORPH in self._seen_attrs and not doc.has_annotation("MORPH"):
-            raise ValueError(Errors.E155.format(pipe="morphologizer", attr="MORPH"))
-        if LEMMA in self._seen_attrs and not doc.has_annotation("LEMMA"):
-            raise ValueError(Errors.E155.format(pipe="lemmatizer", attr="LEMMA"))
-        if DEP in self._seen_attrs and not doc.has_annotation("DEP"):
-            raise ValueError(Errors.E156.format())
+        if not suppress_errors:
```
Comment (honnibal) [inline spacy/matcher/matcher.pyx:218]:
```suggestion
        if not allow_missing:
```

File: spacy/pipeline/attributeruler.py [hunk 3]
```diff
@@ -79,7 +79,7 @@ def __call__(self, doc: Doc) -> Doc:
 
         DOCS: https://nightly.spacy.io/api/attributeruler#call
         """
-        matches = sorted(self.matcher(doc))
+        matches = sorted(self.matcher(doc, suppress_errors=True))
```
Comment (honnibal) [inline spacy/pipeline/attributeruler.py:82]:
```suggestion
        matches = sorted(self.matcher(doc, allow_missing=True))
```

File: spacy/tests/matcher/test_matcher_api.py [hunk 4]
```diff
@@ -316,6 +316,9 @@ def test_attr_pipeline_checks(en_vocab):
         matcher(doc2)
     with pytest.raises(ValueError):
         matcher(doc3)
+    # errors can be suppressed if desired
+    matcher(doc2, suppress_errors=True)
```
Comment (honnibal) [inline spacy/tests/matcher/test_matcher_api.py:320]:
```suggestion
    matcher(doc2, allow_missing=True)
```

File: spacy/tests/matcher/test_matcher_api.py [hunk 5]
```diff
@@ -316,6 +316,9 @@ def test_attr_pipeline_checks(en_vocab):
...
+    matcher(doc3, suppress_errors=True)
```
Comment (honnibal) [inline spacy/tests/matcher/test_matcher_api.py:321]:
```suggestion
    matcher(doc3, allow_missing=True)
```

File: spacy/matcher/matcher.pyx [hunk 6]
```diff
@@ -215,16 +215,17 @@ cdef class Matcher:
...
+            if TAG in self._seen_attrs and not doc.has_annotation("TAG"):
+                raise ValueError(Errors.E155.format(pipe="tagger", attr="TAG"))
+            if POS in self._seen_attrs and not doc.has_annotation("POS"):
+                raise ValueError(Errors.E155.format(pipe="morphologizer", attr="POS"))
+            if MORPH in self._seen_attrs and not doc.has_annotation("MORPH"):
+                raise ValueError(Errors.E155.format(pipe="morphologizer", attr="MORPH"))
+            if LEMMA in self._seen_attrs and not doc.has_annotation("LEMMA"):
+                raise ValueError(Errors.E155.format(pipe="lemmatizer", attr="LEMMA"))
+            if DEP in self._seen_attrs and not doc.has_annotation("DEP"):
+                raise ValueError(Errors.E156.format())
```
Comment (honnibal) [inline spacy/matcher/matcher.pyx:228]: We should probably make this a loop:

```
for attr in [TAG, POS, MORPH, LEMMA, DEP]:
    if attr in self._seen_attrs and not doc.has_annotation(attr):
        raise ValueError(Errors.E155.format(pipe="morphologizer", attr=self.strings[attr]))
```

Comment (adrianeboyd) [reply spacy/matcher/matcher.pyx:228]: Yeah. (This is from when the is_flags weren't easily derivable from the features and they were the same for MORPH/POS.)

**Pull request merged**